packages/xo-web/src/xo-app/dashboard/health/index.js

```diff
@@ -585,6 +604,20 @@ export default class Health extends Component {
     )
   )
 
+  _getDefaultLocalSrs = createCollectionWrapper(
+    createSelector(
+      () => this.props.hosts,
+      () => this.props.pools,
+      () => this.props.userSrs,
+      () => this._getPoolIds(),
+      (hosts, pools, userSrs, poolIds) =>
+        filter(
+          filter(isEmpty(poolIds) ? pools : pick(pools, poolIds), pool => !userSrs[pool.default_SR].shared),
+          pool => countBy(hosts, host => host.$pool)[pool.id] > 1
```

Comment:
The `countBy` will be run again and again every time the callback is called.

Reply:
Why not filtering on the 2 conditions ("number of hosts > 1" and "default SR is local") in the same filter?